assembl/tasks/common.py

```diff
@@ -311,27 +337,334 @@ def create_venv(c, path=None):
+    return 'assembl-backup.pgdump'
+
+
+def remote_db_path(c):
+    return join(c.config.projectpath, get_db_dump_name())
+
+
+def supervisor_process_stop(c, process_name):
+    """
+    Assuming the supervisord process is running, stop one of its processes
+    """
+    print('Asking supervisor to stop %s' % process_name)
+    supervisor_pid_regex = re.compile(r'^\d+')
+    status_regex = re.compile(r'^%s\s*(\S*)' % process_name)
+    with venv(c):
+        supervisord_cmd_result = c.run("supervisorctl pid", warn=True, hide='out')
+    match = supervisor_pid_regex.match(supervisord_cmd_result)
+    if not match:
+        print('Supervisord doesn\'t seem to be running, nothing to stop')
+        return
+    for try_num in range(20):
+        with venv(c):
+            c.run("supervisorctl stop %s" % process_name)
+            status_cmd_result = c.run("supervisorctl status %s" % process_name, hide='out')
+
+        match = status_regex.match(status_cmd_result)
+        if match:
+            status = match.group(1)
+            if(status == 'STOPPED'):
+                print("%s is stopped" % process_name)
+                break
+            if(status == 'FATAL'):
+                print("%s had a fatal error" % process_name)
+                break
+            elif(status == 'RUNNING'):
+                with venv(c):
+                    c.run("supervisorctl stop %s" % process_name)
+            elif(status == 'STOPPING'):
+                print(status)
+            else:
+                print("unexpected status: %s" % status)
+            sleep(1)
+        else:
+            print('Unable to parse status (bad regex?)')
+            print(status_cmd_result)
+            exit()
+
+
+def is_supervisord_running(c):
+    with venv(c):
+        result = c.run('supervisorctl pid')
+    if 'no such file' in result:
+        return False
+    try:
+        pid = int(result)
+        if pid:
+            return True
+    except:
+        return False
+
+
+def webservers_start(c):
     """
-    Adds the SSH private key of the bluenove-bot.
-    In the CI environment, comes as ENV variable. Can be overriden by passing location of private key as an arg.
+    Start all webservers
     """
-    if private_key:
-        if exists(private_key):
-            c.run("ssh-add %s" % private_key)
+    if c.config._internal.uses_nginx:
+        # Nginx
+        if exists(c, '/etc/init.d/nginx'):
+            # Have to ensure that the env.user has visudo rights to call this
+            c.sudo('/etc/init.d/nginx start')
+        elif c.config.mac and exists('/usr/local/nginx/sbin/nginx'):
+            c.sudo('/usr/local/nginx/sbin/nginx')
+
+
+def is_supervisor_running(c):
+    with venv(c):
+        supervisord_cmd_result = c.run("supervisorctl avail", hide='both', warn=True)
+        if supervisord_cmd_result.failed:
+            return False
         else:
-            print("The provided key was not found!")
+            return True
+
+
+def app_reload(c):
```

Comment:
The logic of this function, in `common.py` is not well designed. It's no longer global like this. If this was for development (on mac.py) then it would make sense sort of, but in common, possible to use in production as well, then you'll need to separate this install reload based on what is the supervisor process state of the machine. Ie, in ASG boxes, prod:uwsgi is on. On celery, it `celery` and `celery_notify_beat`. In dev mode, it's likely the autostart processes variable. Alas, this function's logic is too outdated for current setup.

Reply:
It s currently only used for development in mac.py. If we wanted to do a global function, could we use the YAML file to determine which supervisor process is running ?